src/java_bytecode/java_bytecode_convert_method.cpp

```diff
@@ -2319,7 +2319,8 @@ codet java_bytecode_convert_methodt::convert_instructions(
     if(!start_new_block && previous_address!=-1)
     {
       for(const auto &exception_row : method.exception_table)
-        if(exception_row.start_pc==previous_address)
+        if(static_cast<decltype(previous_address)>(exception_row.start_pc)==
+           previous_address)
```

Comment:
How about changing the type of `previous_address`, and add adding a separate `bool`-typed variable to track whether the address has been set? As is, assigning `address` to `previous_address` isn't actually safe either.